docs/datamodel/build-a-datamodel.rst

```diff
@@ -0,0 +1,627 @@
+    #: Serializer for search results.
+    json_v1_search = search_responsify(json_v1, 'application/json')
+
+
+First, we create an instance of the ``JSONSerializer`` and provide it with
+our previously created Marshmallow schema. The marshmallow schema is used to
+transform the internal JSON prior to that the ``JSONSerializer`` dumps the
+actual JSON output. This allows you e.g. to evolve your internal data model,
+without affecting your REST API.
+
+Next, we create two different **response serializers**: ``json_v1_response``
+and ``json_v1_search``. The former is responsible for producing an HTTP
+response for an individual record, while the latter is responsible for
+producing an HTTP response for a search result (i.e. multiple records).
+
+The response serializer can not only output data to the HTTP response body, but can also add HTTP headers (e.g. Link headers).
+
+You can see examples of the output from the two response serializers in
+the Quickstart section: :ref:`display-a-record` and :ref:`search-for-records`.
+
+Define loaders
+--------------
+Think of loaders as the definition of your input formats for records. You only
+need loaders if you plan to allow creation of records via the REST API.
+
+The loaders are responsible for transforming a request payload (external representation) into the internal JSON format.
+
+Loaders are defined in the ``loaders`` module:
+
+.. code-block:: shell
+
+    |-- my_datamodel
+    |   |-- loaders
+    |   |   `-- __init__.py
+
+Loaders are defined in much the same as serializers, and similarly you can
+use the Marshmallow schemas:
+
+.. code-block:: python
+
+    from invenio_records_rest.loaders.marshmallow import \
+        marshmallow_loader
+    from ..marshmallow import MetadataSchemaV1
+
+    json_v1 = marshmallow_loader(MetadataSchemaV1)
+
+Note, that you are not required to use Marshmallow for deserialization, but it
+allows you to use advanced data validation rules on your REST API.
+
+Define templates
+----------------
+In order to display records not only on your REST API, but also provide
+search interface and landing pages for your record you need to provide
+templates that render your records.
+
+You will need two different types of templates:
+
+- Search result template
+- Landing page template
+
+The templates are stored in two different folders (``static`` and
+``templates``):
+
+.. code-block:: shell
+
+    |-- my_datamodel
+    |   |-- static
+    |   |   `-- templates
+    |   |       `-- my_datamodel
+    |   |           `-- results.html
+    |   |-- templates
+    |   |   `-- my_datamodel
+    |   |       `-- record.html
+
+
+**Search result template**
+
+The Invenio search interface is run by a JavaScript application, and thus the
+template is rendered client side in the user's browser. The template uses data
+received by the REST API and thus your REST API must be able to deliver all
+information you would like to render in the template.
```

Comment:
Highlighting that this depends on the serializers you have created/you are using would be something which is too evident to mention?